--- ../src-base/minecraft/cpw/mods/fml/common/InjectedModContainer.java
+++ ../src-work/minecraft/cpw/mods/fml/common/InjectedModContainer.java
@@ -13,6 +13,7 @@
 package cpw.mods.fml.common;
 
 import java.io.File;
+import java.net.URL;
 import java.security.cert.Certificate;
 import java.util.List;
 import java.util.Map;
@@ -201,4 +202,11 @@
     {
         return wrappedContainer.getOwnedPackages();
     }
+
+    @Override
+    public URL getUpdateUrl()
+    {
+        return wrappedContainer.getUpdateUrl();
+    }
+
 }
