--- ../src-base/minecraft/cpw/mods/fml/common/Loader.java
+++ ../src-work/minecraft/cpw/mods/fml/common/Loader.java
@@ -86,6 +86,7 @@
 import cpw.mods.fml.common.versioning.VersionParser;
 import cpw.mods.fml.relauncher.ModListHelper;
 import cpw.mods.fml.relauncher.Side;
+import net.minecraftforge.common.config.ConfigManager;
 
 /**
  * The loader class performs the actual loading of the mod code from disk.
@@ -509,6 +510,7 @@
                 }
             }
         }
+        ConfigManager.loadData(discoverer.getASMTable());
         modController.transition(LoaderState.CONSTRUCTING, false);
         modController.distributeStateMessage(LoaderState.CONSTRUCTING, modClassLoader, discoverer.getASMTable(), reverseDependencies);
 
