--- ../src-base/minecraft/cpw/mods/fml/common/Mod.java
+++ ../src-work/minecraft/cpw/mods/fml/common/Mod.java
@@ -144,18 +144,18 @@
      * @return The language the mod is authored in
      */
     String modLanguage() default "java";
-    
+
     /**
      * The language adapter to be used to load this mod. This overrides the value of modLanguage. The class must have a
      * public zero variable constructor and implement {@link ILanguageAdapter} just like the Java and Scala adapters.
-     * 
+     *
      * A class with an invalid constructor or that doesn't implement {@link ILanguageAdapter} will throw an exception and
      * halt loading.
-     * 
+     *
      * @return The full class name of the language adapter
      */
     String modLanguageAdapter() default "";
-    
+
     /**
      * NOT YET IMPLEMENTED. </br>
      * An optional ASM hook class, that can be used to apply ASM to classes loaded from this mod. It is also given
@@ -185,7 +185,15 @@
      * @return The name of a class implementing {@link IModGuiFactory}
      */
     String guiFactory() default "";
+
     /**
+     * An optional URL to a JSON file that will be checked once per launch to determine if there is an updated
+     * version of this mod and notify the end user. For more information see ForgeVersion.
+     * @return URL to update metadata json
+     */
+    String updateJSON() default "";
+
+    /**
      * A list of custom properties for this mod. Completely up to the mod author if/when they
      * want to put anything in here.
      * @return an optional list of custom properties
