--- ../src-base/minecraft/cpw/mods/fml/client/GuiSlotModList.java
+++ ../src-work/minecraft/cpw/mods/fml/client/GuiSlotModList.java
@@ -14,8 +14,15 @@
 
 import java.util.ArrayList;
 
-import net.minecraft.client.renderer.Tessellator;
+import org.lwjgl.opengl.GL11;
 
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.Gui;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.ForgeVersion;
+import net.minecraftforge.common.ForgeVersion.CheckResult;
 import cpw.mods.fml.common.Loader;
 import cpw.mods.fml.common.LoaderState.ModState;
 import cpw.mods.fml.common.ModContainer;
@@ -26,12 +33,14 @@
  */
 public class GuiSlotModList extends GuiScrollingList
 {
+    private static final ResourceLocation VERSION_CHECK_ICONS = new ResourceLocation("fml", "textures/gui/version_check_icons.png");
+
     private GuiModList parent;
     private ArrayList<ModContainer> mods;
 
-    public GuiSlotModList(GuiModList parent, ArrayList<ModContainer> mods, int listWidth)
+    public GuiSlotModList(GuiModList parent, ArrayList<ModContainer> mods, int listWidth, int slotHeight)
     {
-        super(parent.getMinecraftInstance(), listWidth, parent.field_146295_m, 32, parent.field_146295_m - 66 + 4, 10, 35);
+        super(parent.getMinecraftInstance(), listWidth, parent.field_146295_m, 32, parent.field_146295_m - 88 + 4, 10, slotHeight, parent.field_146294_l, parent.field_146295_m);
         this.parent=parent;
         this.mods=mods;
     }
@@ -66,21 +75,37 @@
         return (this.getSize()) * 35 + 1;
     }
 
+    ArrayList<ModContainer> getMods()
+    {
+        return mods;
+    }
+
     @Override
     protected void drawSlot(int listIndex, int var2, int var3, int var4, Tessellator var5)
     {
         ModContainer mc=mods.get(listIndex);
+        FontRenderer font = this.parent.getFontRenderer();
+        CheckResult  vercheck = ForgeVersion.getResult(mc);
         if (Loader.instance().getModState(mc)==ModState.DISABLED)
         {
             this.parent.getFontRenderer().func_78276_b(this.parent.getFontRenderer().func_78269_a(mc.getName(), listWidth - 10), this.left + 3 , var3 + 2, 0xFF2222);
-            this.parent.getFontRenderer().func_78276_b(this.parent.getFontRenderer().func_78269_a(mc.getDisplayVersion(), listWidth - 10), this.left + 3 , var3 + 12, 0xFF2222);
+            this.parent.getFontRenderer().func_78276_b(this.parent.getFontRenderer().func_78269_a(mc.getDisplayVersion(), listWidth - (5 + var4)), this.left + 3 , var3 + 12, 0xFF2222);
             this.parent.getFontRenderer().func_78276_b(this.parent.getFontRenderer().func_78269_a("DISABLED", listWidth - 10), this.left + 3 , var3 + 22, 0xFF2222);
         }
         else
         {
             this.parent.getFontRenderer().func_78276_b(this.parent.getFontRenderer().func_78269_a(mc.getName(), listWidth - 10), this.left + 3 , var3 + 2, 0xFFFFFF);
-            this.parent.getFontRenderer().func_78276_b(this.parent.getFontRenderer().func_78269_a(mc.getDisplayVersion(), listWidth - 10), this.left + 3 , var3 + 12, 0xCCCCCC);
+            this.parent.getFontRenderer().func_78276_b(this.parent.getFontRenderer().func_78269_a(mc.getDisplayVersion(), listWidth - (5 + var4)), this.left + 3 , var3 + 12, 0xCCCCCC);
             this.parent.getFontRenderer().func_78276_b(this.parent.getFontRenderer().func_78269_a(mc.getMetadata() !=null ? mc.getMetadata().getChildModCountString() : "Metadata not found", listWidth - 10), this.left + 3 , var3 + 22, 0xCCCCCC);
+
+            if (vercheck.status.shouldDraw())
+            //TODO: Consider adding more icons for visualization
+            Minecraft.func_71410_x().func_110434_K().func_110577_a(VERSION_CHECK_ICONS);
+            GL11.glClearColor(1, 1, 1, 1);
+            GL11.glPushMatrix();
+            // func_146110_a = drawModalRectWithCustomSizedTexture
+            Gui.func_146110_a(right - (var4 / 2 + 4), top + (var4 / 2 - 4), vercheck.status.getSheetOffset() * 8, (vercheck.status.isAnimated() && ((System.currentTimeMillis() / 800 & 1)) == 1) ? 8 : 0, 8, 8, 64, 16);
+            GL11.glPopMatrix();
         }
     }
 
